### ISSUE TITLE
FR - directly define authority/token eps with tenant id in config.yml 

### ISSUE THREAD

**Issue by pandel** [SMTP2Graph/SMTP2Graph#16]
Hi!

I was trying to test SMTP2Graph in a very restricted company network and found out, that it is not possible to acquire the authority and token endpoints via the tenant name like it is done in `Mailer.ts` atm via

```
authority: `https://login.microsoftonline.com/${Config.clientTenant}.onmicrosoft.com`,
```

according to this error message in the log:

```
{"error":"endpoints_resolution_error: Endpoints cannot be resolved","filename":"nicmdqos5k6ey4co.eml","level":"error","message":"[MailQueue] Failed to send message \"nicmdqos5k6ey4co.eml\"","name":"ClientAuthError","stack":"ClientAuthError: endpoints_resolution_error: Endpoints cannot be resolved\n    at os (dist/server.js:2:867462)\n    at Br.createDiscoveredInstance (dist/server.js:2:934155)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async pa.value.f.clientId.auth.authority.buildOauthClientConfiguration (dist/server.js:2:991352)\n    at async pa.value.f.clientId.auth.authority.acquireTokenByClientCredential (dist/server.js:2:1007400)\n    at async dist/server.js:2:1006373","timestamp":"2024-06-27T11:47:57+02:00"}
```

In another, but commercial application running in this environment the vendor had to make the OAuth endpoints definable directly (aka via the tenant ID):

(anonymized)
```
OAuth 2.0 authorization endpoint (v2):   https://login.microsoftonline.com/XXXXXXXX-BBBB-4019-KKKK-FFFFFFFFFFFF/oauth2/v2.0/authorize
OAuth 2.0 token endpoint (v2):   https://login.microsoftonline.com/XXXXXXXX-BBBB-4019-KKKK-FFFFFFFFFFFF/oauth2/v2.0/token
```

Would this be possible here, too?

Regards,
Holger

**Comment by RoelVB** [comment]
Interesting. I have never seen this.
I will add support for using the Tenant ID instead of the name.

**Comment by RoelVB** [comment]
Please try version 1.0.3 and enter your tenantID into the config instead of the tenant name

**Comment by pandel** [comment]
Many thanks for implementing this. It did in fact solve the issue, that the Tenant ID wasn't resolvable (I tried manually in the browser), but we noticed another fact, that is beyond the scope of this problem (and I am so sorry, I did not know about it until now): generally, we are using a corporate proxy, but our last information about this proxy was, that MSO traffic explicitly would not be routed through this proxy - this is not correct as I do know now after researching this problem here... as SMTP2Graph isn't picking up any proxy setting defined in the underlying OS, the outgoing connection to the authorization endpoints is unsuccessful.

So, I feel very sorry having bothered you with my problem! As I said, if I had know this before, I would not have posted the issue... 

FWIW, regarding the proxy problem, ConfidentCredentialClient and a possible solution, see: https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/6527
